command_line_assistant/utils/cli.py

```diff
@@ -31,6 +34,15 @@
 
 OS_RELEASE_PATH = Path("/etc/os-release")
 
+# Type definitions
+CommandFunc = Callable[[Namespace, "CommandContext"], int]
+ArgumentSpec = tuple[tuple[str, ...], dict[str, Any]]
+
+# Global storage for registered commands
+_commands: dict[str, "Command"] = {}
```

Comment:
this says its global but also uses an underscore (which typically indicates internal). I'm not sure which is correct